src/ecdsa.rs

```diff
@@ -55,3 +55,144 @@ impl ToPyObject for PySignature {
         self.clone().into_py(py)
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::relocatable::PyRelocatable;
+    use num_bigint::{BigInt, Sign};
+
+    use crate::cairo_runner::PyCairoRunner;
+
+    use std::fs;
+
+    #[test]
+    fn create_empty_py_signature() {
+        PySignature::new();
+    }
+
+    #[test]
+    fn add_py_signature() {
+        let rel = PyRelocatable {
+            segment_index: 2,
+            offset: 0,
+        };
+
+        let numbers = (
+            BigInt::new(Sign::Plus, vec![1, 0, 0, 0, 0, 0, 17, 134217728]),
+            BigInt::new(Sign::Plus, vec![1, 0, 0, 0, 0, 0, 17, 134217728]),
+        );
+
+        let mut signature = PySignature::new();
+
+        signature.add_signature(rel, numbers);
+    }
+
+    #[test]
+    fn update_py_signature() {
+        let rel = PyRelocatable {
+            segment_index: 2,
+            offset: 0,
+        };
+
+        let numbers = (
+            BigInt::new(Sign::Plus, vec![1, 0, 0, 0, 0, 0, 17, 13421772]),
+            BigInt::new(Sign::Plus, vec![1, 0, 0, 0, 0, 0, 17, 13421772]),
+        );
+
+        let mut signature = PySignature::new();
+        let original_signature = signature.clone();
+
+        signature.add_signature(rel, numbers);
+
+        let path = "cairo_programs/ecdsa.json".to_string();
+        let program = fs::read_to_string(path).unwrap();
+        let mut runner = PyCairoRunner::new(
+            program,
+            Some("main".to_string()),
+            Some("all".to_string()),
+            false,
+        )
+        .unwrap();
+
+        match runner.initialize() {
+            Ok(_v) => assert!(true),
+            Err(e) => assert!(false, "error: {e:?}"),
+        }
```

Comment:
We could use unwrap or expect here
```suggestion
        runner.initialize().expect("Failed to initialize VM")
```

Reply:
done!